concepts/05 UI Components/DataGrid/10 Enhance Performance on Large Datasets/030 Rendering Optimization.md

```diff
@@ -1,7 +1,16 @@
+If you work with a large data set, apply the following recommendations to optimize rendering performance:
 
-Use virtual row rendering when [paging is disabled](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/enabled.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#enabled') or the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') is more than the default value.
+* **Render only visible rows and columns**             
+The DataGrid renders all rows and columns once the data is loaded. To optimize performance, set the [rowRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/rowRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#rowRenderingMode') and [columnRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/columnRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#columnRenderingMode') properties in the **scrolling** object to *"virtual"*. These properties specify that UI elements are only rendered when they come into the viewport. Use virtual row rendering when [paging is disabled](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/enabled.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#enabled') or the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') is more than the default value. Virtual column rendering should be used when there are more than 20 columns outside the viewport.
 
-Virtual column rendering can be used when there are more than 20 columns outside the viewport.
+* **Render elements with complex content last**           
+Cells with complex content (filter row, command columns, editors and columns with checkboxes) can be rendered after cells with simple content. To do that, enable the [renderAsync](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync') property. In addition, you can enable [the same property](/api-reference/_hidden/GridBaseColumn/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync') for individual columns if they use a content-heavy [cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate').
```

Comment:
The first sentence can be turned into active form. This will bring the subject and verb together at the beginning of the sentence for clearer flow. 

There's not explicit indication of the effect that this rendering strategy has on end-users. You might want to add a sentence in the beginning of after the first one that explains why that matters. I could be clear to many, but still it makes sense to state your point exactly, so that no doubt remains. 

'They use a cellTemplate' - you go from plural to singular (probably to match the property name). If it plural "columns" then it should be "templates". So I changed that part a bit, including the link text. 

```suggestion
The grid control can render cells with simple content first, and only then proceed to cells with complex content (filter row, command columns, editors and columns with checkboxes). If the control does this, end-users start to see content sooner and thus perceive the app UI as more responsive. To load simple content first, use the [renderAsync](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync') property. In addition, you can enable [the same property](/api-reference/_hidden/GridBaseColumn/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync') for individual columns if they use content-heavy [cell templates](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate').
```